### PR TITLE
nghttp2: add 1.56.0

### PR DESCRIPTION
**Pull request by hrxi**

### CODE REVIEW

**Comment by hrxi** [comment]
Known build problem on MSVC: https://github.com/nghttp2/nghttp2/issues/1633. What should I do? Ignore the build failures?

Ignoring them in `ci_config.json` doesn't work, it expects the meson setup call to fail.

**Comment by neheb** [comment]
Define it with add_project_arguments?

**Comment by hrxi** [comment]
> Define it with add_project_arguments?

I think that wouldn't propagate to dependents, but they require it for the header. I guess I need to add it to `declare_dependency`, too.

File: subprojects/packagefiles/nghttp2/lib/meson.build [hunk 1]
```diff
@@ -0,0 +1,63 @@
+subdir('includes')
+configure_file(
```
Comment (neheb) [inline subprojects/packagefiles/nghttp2/lib/meson.build:2]: IMO use the regular pkgconfig module instead of patching the pc file.

**Comment by neheb** [comment]
yes.

Windows has a SSIZE_T in BaseTsd.h FWIW. Maybe that can be adapted for upstream.

**Comment by neheb** [comment]
@hrxi new version available. Also, this doesn't work with newer clang where implicit declarations become errors.

**Pull request merged**